### PR TITLE
feat(redux|client): add additional user attributes client

### PR DESCRIPTION
**Pull request by andreecosta**
## Description

- Add additional user attributes client
- Converted all files inside users folder to TypeScript for both client and redux, which required to type methods or variables that were not typed until now.

### Dependencies

None

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Comment by joaoprcosta** [comment]
Missing the description in the commit message.

**Comment by nelsonleite** [comment]
It seems to have a lot more changes than just adding a client... Can you please describe what's actually being done here? 

(And things like "to next" are irrelevant, the branch itself shows us that 🙂 )

File: packages/client/src/users/__tests__/getUserAttribute.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  beforeEach(() => {
+    moxios.install(client);
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => moxios.uninstall(client));
+
+  it('should handle a client request successfully', async () => {
+    const response: UserAttributesResponse = {
```
Comment (jhairabreu) [inline packages/client/src/users/__tests__/getUserAttribute.test.ts:21]: I think these mock responses could be added to the general mock files that we have (`tests/__fixtures__/users/users.fixtures.ts`), and then import it from there, don't you think? 
You can check other areas to see how we've done it.

Also check other if this applies to the other files you changed 🙂

File: packages/client/src/users/__tests__/postUserAttributes.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  beforeEach(() => {
+    moxios.install(client);
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => moxios.uninstall(client));
+
+  it('should handle a client request successfully', async () => {
+    const response: UserAttributesResponse = {
```
Comment (jhairabreu) [inline packages/client/src/users/__tests__/postUserAttributes.test.ts:30]: same.

**Review by jhairabreu — CHANGES_REQUESTED** [review]

File: packages/client/src/users/__tests__/getUserAttribute.test.ts (hunk 1 above)
Comment (andreecosta) [reply packages/client/src/users/__tests__/getUserAttribute.test.ts]: done :D

File: packages/client/src/users/__tests__/postUserAttributes.test.ts (hunk 2 above)
Comment (andreecosta) [reply packages/client/src/users/__tests__/postUserAttributes.test.ts]: and done :)

**Review by jhairabreu — APPROVED** [review]

**Review by pedro-gbf — APPROVED** [review]

**Pull request merged**